code/modules/reagents/grenade_launcher.dm

```diff
@@ -49,18 +54,49 @@
 		to_chat(usr, "<span class='warning'>The [src.name] is empty.</span>")
 
 /obj/item/weapon/gun/grenadelauncher/proc/fire_grenade(atom/target, mob/user)
-	for(var/mob/O in viewers(world.view, user))
-		O.show_message(text("<span class='warning'>[] fired a grenade!</span>", user), 1)
+	user.visible_message("<span class='warning'>[user] fired a grenade!</span>")
 	to_chat(user, "<span class='warning'>You fire the grenade launcher!</span>")
-	var/obj/item/weapon/grenade/chem_grenade/F = grenades[1] //Now with less copypasta!
+	var/obj/item/weapon/grenade/F = grenades[1] //Now with less copypasta!
 	grenades -= F
 	F.forceMove(user.loc)
 	F.throw_at(target, 30, 2)
 	message_admins("[key_name_admin(user)] fired [F.name] from [src.name].")
 	log_game("[key_name_admin(user)] launched [F.name] from [src.name].")
 	playsound(user.loc, 'sound/weapons/grenadelauncher.ogg', 50, 1, -3)
+	if(timer_override)
+		F.det_time = timer_override_value
 	F.activate()
 
+/obj/item/weapon/gun/grenadelauncher/verb/TimerOverrideMode(mob/user as mob)
+	set name = "Toggle Timer Override Mode"
+	set category = "Object"
+	set src in usr
+
+	if(timer_override)
+		to_chat(user, "You deactivate the timing override on \the [src]. Fired grenades will now use their individual timers")
+		timer_override = FALSE
+	else
+		var/new_time = input(user, "Fired grenades will now be prime using \the [src]'s timer rather than their own. What would you like the timer to be set to, in seconds?","Time",timer_override_value/10) as num
+		if(new_time < 3)
+			to_chat(user, "<span class = 'warning'>\The [src] beeps. The given time cannot be below 3 second.</span>")
+			return
+		if(!Adjacent(user, MAX_ITEM_DEPTH))
+			to_chat(user, "<span class = 'warning'>You fumble while trying to modify \the [src]. It is best to keep the [src] in proximity when modifying it..</span>")
+			return
+		timer_override_value = new_time SECONDS
+		to_chat(user, "<span class = 'notice'>The detonation time on your shots is now [timer_override_value/10] seconds.</span>")
+		timer_override = TRUE
+	return
+
+
+/obj/item/weapon/gun/grenadelauncher/verb/Unload(mob/user )
+	set name = "Unload Grenades"
+	set category = "Object"
+	set src in usr
+	for(var/obj/item/weapon/grenade/G in grenades)
+		grenades -= G
+		G.forceMove(user.loc)
```

Comment:
check to see if the user is on a turf first, to prevent unloading grenades into unreachable locations such as inside a mech or sleeper

Reply:
Address then mergetown